tests/test_job.py

```diff
@@ -123,6 +123,92 @@ def test_list(self, connection, add_response, limit, ids, want_params, want_name
         have_params = responses.calls[0].request.params  # pyright: reportGeneralTypeIssues=false
         assert have_params == want_params
 
+    @pytest.mark.parametrize(
+        "limit, ids, status, want_params, want_names",
+        [
+            (
+                1,
+                None,
+                "queued",
+                {"size": "1", "status": "queued"},
+                ["foo"],
+            ),
+            (
+                2,
+                None,
+                None,
+                {"size": "2"},
+                ["foo", "bar"],
+            ),
+            (
+                2,
+                None,
+                "invalid",
+                {"size": "2", "status": "invalid"},
+                [],
+            ),
+            (
+                2,
+                [
+                    "00000000-0000-0000-0000-000000000001",
+                    "00000000-0000-0000-0000-000000000002",
+                    "00000000-0000-0000-0000-000000000003",
+                ],
+                "complete",
+                {
+                    "size": "3",
+                    "id": [
+                        "00000000-0000-0000-0000-000000000001",
+                        "00000000-0000-0000-0000-000000000002",
+                        "00000000-0000-0000-0000-000000000003",
+                    ],
+                    "status": "complete",
+                },
+                ["bar"],
+            ),
+        ],
+    )
+    @responses.activate
+    def test_list_status(
+        self, connection, add_response, limit, ids, status, want_params, want_names
+    ):
+        """Tests that the correct jobs are listed and that the correct query
+        parameters are encoded in the HTTP request to the Xanadu Cloud platform.
+        We ensure that the jobs are filtered by the given status.
```

Comment:
Our convention is to write docstrings in third-person:
```suggestion
        """Tests that the correct jobs are listed and that the correct status
        parameter is encoded in the HTTP request to the Xanadu Cloud platform.
```